packages/suite/src/middlewares/suite/logsMiddleware.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MiddlewareAPI } from 'redux';
+import { AppState, Action, Dispatch } from '@suite-types';
+import * as logActions from '@suite-actions/logsActions';
+import { TRANSPORT, DEVICE } from '@trezor/connect';
+import {
+    ANALYTICS,
+    DESKTOP_UPDATE,
+    METADATA,
+    MODAL,
+    PROTOCOL,
+    ROUTER,
+    SUITE,
+} from '@suite-actions/constants';
+import { ACCOUNT, DISCOVERY, BLOCKCHAIN } from '@wallet-actions/constants';
+import { getAccountIdentifier } from '@suite/utils/wallet/accountUtils';
+import { WALLET_SETTINGS } from '@settings-actions/constants';
+import { redactTransactionIdFromAnchor } from '@suite-utils/analytics';
+
+const log =
+    (api: MiddlewareAPI<Dispatch, AppState>) =>
+    (next: Dispatch) =>
+    (action: Action): Action => {
+        next(action);
+
+        // avoid endless loops, see default in switch
+        // also do not log any log related actions
+        if (action.type.startsWith('@log')) return action;
+
+        switch (action.type) {
+            case SUITE.SET_LANGUAGE:
+            case SUITE.SET_THEME:
+            case SUITE.SET_AUTODETECT:
+            case WALLET_SETTINGS.SET_LOCAL_CURRENCY:
+            case WALLET_SETTINGS.SET_HIDE_BALANCE:
+            case METADATA.ENABLE:
+            case METADATA.DISABLE:
+            case SUITE.ONION_LINKS:
+            case ANALYTICS.ENABLE:
+            case ANALYTICS.DISABLE:
+            case DESKTOP_UPDATE.CHECKING:
+            case DESKTOP_UPDATE.AVAILABLE:
+            case DESKTOP_UPDATE.NOT_AVAILABLE:
+            case DESKTOP_UPDATE.READY:
+            case MODAL.CLOSE:
+                api.dispatch(
+                    logActions.addAction(action, {
+                        ...action,
+                        type: undefined,
+                    }),
+                );
+                break;
+            case SUITE.AUTH_DEVICE:
+            case DEVICE.CONNECT:
+            case DEVICE.DISCONNECT:
+            case ACCOUNT.CREATE:
+            case ACCOUNT.UPDATE:
+            case DISCOVERY.COMPLETE:
+            case SUITE.UPDATE_SELECTED_DEVICE:
+            case SUITE.REMEMBER_DEVICE:
+                api.dispatch(
+                    logActions.addAction(action, {
+                        ...action.payload,
+                        type: undefined,
```

Comment:
Why setting to `undefined`?

Reply:
Probably copy-paste error from `addAction` above. Thank you. 066b10f405692870496c969745a0a63ba5771aca

Reply:
and why is it set to undefined in the other `addAction`s?

Reply:
because I spread `action` which contains `type`

Those actions do not contain data in `payload` but on "root" level